File: frontend/src/controllers/userController.ts
import { ISignupInputModels } from "../types/inputModels";
import { api } from "../service/api";
import { resolveResponse } from "../utils";
import { IUser } from "../types/models";

const route = 'User'

const Signup = (input: ISignupInputModels) => 
  new Promise<IUser>((resolve,reject)=>
    api.post(`${route}/signup`, input).then(({data})=> resolveResponse<IUser>(resolve,reject,data)).catch(()=>reject("Erro interno")))

const GetAll = () => 
  new Promise<IUser[]>((resolve,reject)=>
    api.get(`${route}/get-all`).then(({data})=> resolveResponse<IUser[]>(resolve,reject,data)).catch(()=>reject("Erro interno")))

const Disabled = (id: string) => 
  new Promise<boolean>((resolve,reject)=>
    api.put(`${route}/delete/${id}`).then(({data})=> resolveResponse<boolean>(resolve,reject,data)).catch(()=>reject("Erro interno")))

export const userController = {
  Signup,
  GetAll,
  Disabled
}

File: backend/Backend.Api/Controllers/UserController.cs
using Backend.Aplication.InputModels;
using Backend.Aplication.Services.Interfaces;
using Backend.Aplication.ViewModels;
using Backend.Domain.Validators;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace Backend.Api.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class UserController : ControllerBase
    {
        private readonly IUserService _service;
        public UserController(IUserService service) => _service = service;

        [HttpPost("signup")]
        public async Task<ResultValidator<UserViewModel>> Signup([FromBody] RegisterInputModel input) =>
            await _service.Signup(input);

        [HttpGet("get-all")]
        public async Task<ResultValidator<List<UserViewModel>>> GetAll() =>
            await _service.GetAll();

        [HttpPut("delete/{id}")]
        public async Task<ResultValidator<bool>> Disable(Guid id) =>
            await _service.Disable(id);
    }
}
